test/test_scan.py::test_scan test appears to be flakey

An incremental CI without a related change https://github.com/ros-infrastructure/rosdoc2/pull/174 failed once and then passed when an unrelated comment was added to the code. 

There's a warning about os.fork() and potential race conditions which this flakey test might indicate. 

```
1 packages failed:
never_doxygen: retval=3: JSONDecodeError Expecting value: line 1 column 1 (char 0)
------------------------------ Captured log call -------------------------------
WARNING  rosdoc2.scan:impl.py:119 never_doxygen (18/21) returned 3: JSONDecodeError Expecting value: line 1 column 1 (char 0)
=============================== warnings summary ===============================
test/test_flake8.py::test_flake8
  /home/runner/work/rosdoc2/rosdoc2/venv/lib/python3.12/site-packages/flake8_import_order/styles.py:3: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    from pkg_resources import iter_entry_points

test/test_flake8.py::test_flake8
  /home/runner/work/rosdoc2/rosdoc2/venv/lib/python3.12/site-packages/pkg_resources/__init__.py:3142: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('sphinxcontrib')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(pkg)

test/test_scan.py: 20 warnings
  /usr/lib/python3.12/multiprocessing/popen_fork.py:66: DeprecationWarning: This process (pid=3209) is multi-threaded, use of fork() may lead to deadlocks in the child.
    self.pid = os.fork()

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED test/test_scan.py::test_scan - AssertionError: output directory /tmp/pytest-of-runner/pytest-0/scan_output should have a subdirectory never_doxygen
assert False
 +  where False = is_dir()
 +    where is_dir = PosixPath('/tmp/pytest-of-runner/pytest-0/scan_output/never_doxygen').is_dir
 +      where PosixPath('/tmp/pytest-of-runner/pytest-0/scan_output/never_doxygen') = joinpath('never_doxygen')
 +        where joinpath = PosixPath('/tmp/pytest-of-runner/pytest-0/scan_output').joinpath
 +        and   'never_doxygen' = PosixPath('test/packages/never_doxygen').name
================== 1 failed, 22 passed, 22 warnings in 35.27s ==================
```

--- a/rosdoc2/verbs/scan/impl.py
+++ b/rosdoc2/verbs/scan/impl.py
@@ -24,12 +24,13 @@ from catkin_pkg.packages import find_packages_allowing_duplicates
 from rosdoc2.verbs.build.impl import main_impl as build_main_impl
 from rosdoc2.verbs.build.impl import prepare_arguments as build_prepare_arguments
 
+mp.set_start_method('spawn', force=True)
+
 logging.basicConfig(
     format='[%(name)s] [%(levelname)s] %(message)s', level=logging.INFO)
 logger = logging.getLogger('rosdoc2')
 logger_scan = logging.getLogger('rosdoc2.scan')
 
-goptions = None
 # Setting the MAX_PACKAGES to a smaller value may be useful in debugging
 # this module, to reduce run time or isolate sections that cause hangs.
 MAX_PACKAGES = 10000
@@ -84,8 +85,7 @@ def prepare_arguments(parser):
 
 def main_impl(options):
     """Execute the program."""
-    global goptions
-    goptions = options
+    import traceback
 
     if options.install_directory is not None:
         logger.warn(
@@ -110,7 +110,7 @@ def main_impl(options):
         logger_scan.info(f'Adding {package.name} for processing')
 
     pool = mp.Pool(maxtasksperchild=1, processes=subprocesses)
-    pool_results = pool.imap_unordered(package_impl, packages)
+    pool_results = pool.imap_unordered(package_impl, ((p, options) for p in packages))
     while True:
         try:
             (package, returns, message) = pool_results.next()
@@ -126,6 +126,7 @@ def main_impl(options):
             break
         except BaseException as e:  # noqa: B902
             logger_scan.error(f'Unexpected error in scan: {type(e).__name__ + " " + str(e)}')
+            print(traceback.format_exc())
             break
     logger_scan.info('Finished')
     # I'd prefer close() then join() but that seems to sometimes hang.
@@ -145,10 +146,10 @@ def _clocktime():
     return time.strftime('%H:%M:%S')
 
 
-def package_impl(package):
+def package_impl(package_options):
     """Execute for a single function."""
-    global goptions
-    options = Struct(**goptions.__dict__)
+    (package, options) = package_options
+    options = Struct(**options.__dict__)
     package_path = os.path.dirname(package.filename)
     options.package_path = package_path
     return_value = 100

